include/mls/x509_cert_helper.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+#include "bytes/bytes.h"
+#include <memory>
+
+using namespace std;
+using namespace bytes_ns;
+
+namespace mls {
+
+// TODO: move this inside X509Certificate to not mess up global space
+enum struct SignatureAlgorithm : uint16_t {
+	unknown = 0x0000,
+	ED25519 = 0x0001,
+	ED448 = 0x0002,
+};
+
+struct X509Certificate {
```

Comment:
Not totally clear to me that you need a struct here, but maybe it's convenient.  Seems like it should just be:

```
struct Certificate {
  explicit Certificate(const bytes& der);
  static std::vector<Certificate> parse_chain(const bytes& der_chain);

  const bytes& raw() const;
  const SignaturePublicKey& public_key() const;
  bool signed_by(const Certificate& parent) const;

  private:
  typed_unique_ptr<X509> _x509;
  bytes _raw;
  SignaturePublicKey _public_key;

  Certificate(X509* x509_in, bytes raw_in);
};
```

... since that's all we need inside of MLS.